### PR TITLE
Kyle

### PR DESCRIPTION
**Pull request by kbricker8**
Profile functionality is implemented with firestore, the default profile is "John Doe". Creating new profiles has been added, as well as functionality to work with the habit activities and methods.

### CODE REVIEW

File: CMPUT301Project/app/src/main/java/com/example/GroupProject/ProfileFragment.java [hunk 1]
```diff
@@ -20,6 +37,18 @@
...
+    private static final String TAG = "ProfileFragment";
+
+    Context thisContext;
+
+    EditText usernameEdit;
+    EditText fnameEdit;
+    EditText lnameEdit;
+    ImageButton confirmUserbtn;
+
+    User thisUser;
```
Comment (schen32) [inline CMPUT301Project/app/src/main/java/com/example/GroupProject/ProfileFragment.java:49]: Have you tested this? I don't see thisUser being initialized anywhere.

File: CMPUT301Project/app/src/main/java/com/example/GroupProject/ProfileFragment.java [hunk 2]
```diff
@@ -31,24 +60,96 @@ public ProfileFragment() {
...
+            }
+            if (fnameEdit.getText() != null) {
+                String newFname = fnameEdit.getText().toString();
+                thisUser.setFirstName(newFname);
+            }
+            if (lnameEdit.getText() != null) {
+                String newLname = lnameEdit.getText().toString();
+                thisUser.setLastName(newLname);
+            }
+            user.put("FirstName", thisUser.getFirstName());
```
Comment (schen32) [inline CMPUT301Project/app/src/main/java/com/example/GroupProject/ProfileFragment.java:98]: What happens when one of the text fields is null and the property is not set? Then there is no value to get.

Comment (schen32) [reply CMPUT301Project/app/src/main/java/com/example/GroupProject/ProfileFragment.java:98]: You might want to disable the button unless all text fields have a value.

File: CMPUT301Project/app/src/main/java/com/example/GroupProject/ProfileFragment.java [hunk 3]
```diff
@@ -31,24 +60,96 @@ public ProfileFragment() {
...
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        if (getArguments() != null) {
-        }
+
+        //username = getArguments().getString("username");
+        username = ((GroupProject) getActivity().getApplication()).getUsername();
+        thisUser = new User(username);
+        thisUser.setFirstName("");
```
Comment (schen32) [inline CMPUT301Project/app/src/main/java/com/example/GroupProject/ProfileFragment.java:114]: Firstname and lastname should probably be initialized in the constructor.

File: CMPUT301Project/app/src/main/java/com/example/GroupProject/User.java [hunk 4]
```diff
@@ -5,21 +5,22 @@
...
-    String firstName;
-    String lastName;
 
-    User(String username, String firstName, String lastName) {
+    private String username;
+    private String firstName;
+    private String lastName;
+
+    User(String username) {
         this.username = username;
```
Comment (schen32) [inline CMPUT301Project/app/src/main/java/com/example/GroupProject/User.java:23]: Initialize first and lastname here. Add back the other constructor for future use.

File: CMPUT301Project/app/src/main/java/com/example/GroupProject/ProfileFragment.java (hunk 3 above)
Comment (kbricker8) [reply CMPUT301Project/app/src/main/java/com/example/GroupProject/ProfileFragment.java:114]: I didn't do that incase someone doesn't want their full name in the app.

File: CMPUT301Project/app/src/main/java/com/example/GroupProject/ProfileFragment.java (hunk 2 above)
Comment (kbricker8) [reply CMPUT301Project/app/src/main/java/com/example/GroupProject/ProfileFragment.java:98]: If username has no value it should revert back to the original username. Meanwhile if first or last name are empty they will be put into the database as empty strings, that would be the case that the user wants privacy and doesnt want their name in the app.

File: CMPUT301Project/app/src/main/java/com/example/GroupProject/ProfileFragment.java [hunk 5]
```diff
@@ -31,24 +60,96 @@ public ProfileFragment() {
...
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        FirebaseFirestore db;
+        db = getFirestoreInstance();
+
+        confirmUserbtn.setOnClickListener(view1 -> {
+            Map<String, Object> user = new HashMap<>();
```
Comment (schen32) [inline CMPUT301Project/app/src/main/java/com/example/GroupProject/ProfileFragment.java:85]: This function is very overloaded. You can add a new user, edit a user, and switch to another user using the same button. We can keep it like this for now, but the functionalities will need to be split in the future.

**Review by schen32 — APPROVED** [review]

**Pull request merged**